### PR TITLE
[BEAM-9286] Create validation runner test for metrics (user counter). 

### PR DESCRIPTION
**Pull request by HuangLED**
Create dedicated validation runner tests for MonitoringInfo based metrics. 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by HuangLED** [comment]
R: ardagan@, ajamato@

**Comment by aaltay** [comment]
cc: @robertwb

File: sdks/python/apache_beam/metrics/metric_test.py [hunk 1]
```diff
@@ -127,6 +134,35 @@ def test_distribution_empty_namespace(self):
...
+    metric_results = (
+      res.metrics().query(MetricsFilter()
+        .with_name('metrics_user_counter_element')))
+    outputs_counter = metric_results['counters'][0]
+    assert_that(results, equal_to([1, 2, 3, 4]))
+
+    self.assertEqual(outputs_counter.key.metric.name,
+					 'metrics_user_counter_element')
+    self.assertEqual(outputs_counter.committed, 4)
+    self.assertEqual(outputs_counter.attempted, 4)
```
Comment (robertwb) [inline sdks/python/apache_beam/metrics/metric_test.py]: >= 4, no guarantee it's exactly 4.

File: sdks/python/apache_beam/metrics/metric_test.py [hunk 2]
```diff
@@ -127,6 +134,35 @@ def test_distribution_empty_namespace(self):
...
+    nums = pipeline | 'Input' >> beam.Create([1, 2, 3, 4])
+    results = nums | 'ApplyPardo' >> beam.ParDo(SomeDoFn())
+
+    res = pipeline.run()
+    res.wait_until_finish()
+    metric_results = (
+      res.metrics().query(MetricsFilter()
+        .with_name('metrics_user_counter_element')))
+    outputs_counter = metric_results['counters'][0]
+    assert_that(results, equal_to([1, 2, 3, 4]))
```
Comment (robertwb) [inline sdks/python/apache_beam/metrics/metric_test.py]: This must be applied before the pipeline is run.

File: sdks/python/apache_beam/metrics/metric_test.py (hunk 1 above)
Comment (HuangLED) [reply sdks/python/apache_beam/metrics/metric_test.py]: removed the assert on attempted.

File: sdks/python/apache_beam/metrics/metric_test.py (hunk 2 above)
Comment (HuangLED) [reply sdks/python/apache_beam/metrics/metric_test.py]: doen.

**Comment by HuangLED** [comment]
> I'm surprised we don't already have a test for such a basic feature.
> 
> FYI, you can use yapf to format your code: https://cwiki.apache.org/confluence/display/BEAM/Python+Tips

Thanks Robert.  I was able to find larger test that cover user counters, but no dedicated validation runner test for metrics. 

PR updated.

**Comment by HuangLED** [comment]
Run PythonFormatter PreCommit

**Comment by yifanzou** [comment]
test this please

**Comment by yifanzou** [comment]
retest this please.

**Comment by yifanzou** [comment]
Run Python PreCommit

**Comment by HuangLED** [comment]
Run Python PreCommit

**Comment by yifanzou** [comment]
Run Python PreCommit

**Pull request merged**